rust/crd/src/lib.rs

```diff
@@ -138,17 +155,32 @@ impl KafkaCluster {
         }
     }
 
+    pub fn role(&self, role_variant: &KafkaRole) -> Result<&Role<KafkaConfigFragment>, Error> {
+        match role_variant {
+            KafkaRole::Broker => self.spec.brokers.as_ref(),
+        }
+        .with_context(|| CannotRetrieveKafkaRoleSnafu {
+            role: role_variant.to_string(),
+        })
+    }
+
     pub fn rolegroup(
         &self,
         rolegroup_ref: &RoleGroupRef<KafkaCluster>,
-    ) -> Option<(&Role<KafkaConfigFragment>, &RoleGroup<KafkaConfigFragment>)> {
-        match rolegroup_ref.role.parse().ok()? {
-            KafkaRole::Broker => {
-                let role = &self.spec.brokers.as_ref()?;
-                let rg = role.role_groups.get(&rolegroup_ref.role_group)?;
-                Some((role, rg))
-            }
-        }
+    ) -> Result<RoleGroup<KafkaConfigFragment>, Error> {
+        let role_variant =
+            KafkaRole::from_str(&rolegroup_ref.role).with_context(|_| UnknownKafkaRoleSnafu {
+                role: rolegroup_ref.role.to_owned(),
+                roles: KafkaRole::roles(),
+            })?;
+
+        let role = self.role(&role_variant)?;
+        role.role_groups
+            .get(&rolegroup_ref.role_group)
+            .with_context(|| CannotRetrieveKafkaRoleGroupSnafu {
+                role_group: rolegroup_ref.role_group.to_owned(),
+            })
+            .cloned()
```

Comment:
Optional: i would prefer to not always clone here, but instead return a reference and let to caller clone if needed

Reply:
Done in dd6de54. Afaik other PRs also clone here.

Reply:
Many of them, yeah :/ The ones I have noticed I did change it though